File: dotnet/src/CarespaceSDK/Configuration/CarespaceConfiguration.cs
using Microsoft.Extensions.Logging;

namespace CarespaceSDK.Configuration;

public class CarespaceConfiguration
{
    public string BaseUrl { get; set; } = "https://api-dev.carespace.ai";
    public string? ApiKey { get; set; }
    public TimeSpan Timeout { get; set; } = TimeSpan.FromSeconds(30);
    public int MaxRetryAttempts { get; set; } = 3;
    public TimeSpan RetryDelay { get; set; } = TimeSpan.FromSeconds(1);
    public string UserAgent { get; set; } = "CarespaceSDK/1.0.0 (.NET)";
    public LogLevel LogLevel { get; set; } = LogLevel.Information;
    public bool EnableRetry { get; set; } = true;
    public bool EnableLogging { get; set; } = true;
    public IDictionary<string, string> DefaultHeaders { get; set; } = new Dictionary<string, string>();

    public static CarespaceConfiguration ForDevelopment(string? apiKey = null)
    {
        return new CarespaceConfiguration
        {
            BaseUrl = "https://api-dev.carespace.ai",
            ApiKey = apiKey,
            LogLevel = LogLevel.Debug
        };
    }

    public static CarespaceConfiguration ForProduction(string apiKey)
    {
        return new CarespaceConfiguration
        {
            BaseUrl = "https://api.carespace.ai",
            ApiKey = apiKey,
            LogLevel = LogLevel.Warning
        };
    }

    public static CarespaceConfiguration ForStaging(string apiKey)
    {
        return new CarespaceConfiguration
        {
            BaseUrl = "https://api-staging.carespace.ai",
            ApiKey = apiKey,
            LogLevel = LogLevel.Information
        };
    }

    public void Validate()
    {
        if (string.IsNullOrWhiteSpace(BaseUrl))
        {
            throw new ArgumentException("BaseUrl is required", nameof(BaseUrl));
        }

        if (!Uri.TryCreate(BaseUrl, UriKind.Absolute, out _))
        {
            throw new ArgumentException("BaseUrl must be a valid URI", nameof(BaseUrl));
        }

        if (Timeout <= TimeSpan.Zero)
        {
            throw new ArgumentException("Timeout must be greater than zero", nameof(Timeout));
        }

        if (MaxRetryAttempts < 0)
        {
            throw new ArgumentException("MaxRetryAttempts cannot be negative", nameof(MaxRetryAttempts));
        }

        if (RetryDelay < TimeSpan.Zero)
        {
            throw new ArgumentException("RetryDelay cannot be negative", nameof(RetryDelay));
        }
    }
}

File: temp-repos/carespace-kotlin-sdk/carespace-sdk/src/commonMain/kotlin/com/carespace/sdk/configuration/CarespaceConfiguration.kt
package com.carespace.sdk.configuration

import io.ktor.client.plugins.logging.*
import kotlin.time.Duration
import kotlin.time.Duration.Companion.seconds

/**
 * Configuration for the Carespace SDK
 */
data class CarespaceConfiguration(
    val baseUrl: String = "https://api-dev.carespace.ai",
    val apiKey: String? = null,
    val timeout: Duration = 30.seconds,
    val maxRetryAttempts: Int = 3,
    val retryDelay: Duration = 1.seconds,
    val enableRetry: Boolean = true,
    val enableLogging: Boolean = true,
    val logLevel: LogLevel = LogLevel.INFO,
    val userAgent: String = "CarespaceSDK/1.0.0 (Kotlin)",
    val defaultHeaders: Map<String, String> = emptyMap()
) {
    companion object {
        /**
         * Create configuration for development environment
         */
        fun forDevelopment(apiKey: String? = null): CarespaceConfiguration {
            return CarespaceConfiguration(
                baseUrl = "https://api-dev.carespace.ai",
                apiKey = apiKey,
                logLevel = LogLevel.ALL
            )
        }

        /**
         * Create configuration for production environment
         */
        fun forProduction(apiKey: String): CarespaceConfiguration {
            return CarespaceConfiguration(
                baseUrl = "https://api.carespace.ai",
                apiKey = apiKey,
                logLevel = LogLevel.NONE
            )
        }

        /**
         * Create configuration for staging environment
         */
        fun forStaging(apiKey: String): CarespaceConfiguration {
            return CarespaceConfiguration(
                baseUrl = "https://api-staging.carespace.ai",
                apiKey = apiKey,
                logLevel = LogLevel.INFO
            )
        }
    }

    /**
     * Validate the configuration
     */
    fun validate() {
        require(baseUrl.isNotBlank()) { "Base URL cannot be blank" }
        require(timeout.isPositive()) { "Timeout must be positive" }
        require(maxRetryAttempts >= 0) { "Max retry attempts cannot be negative" }
        require(retryDelay.isPositive()) { "Retry delay must be positive" }
        require(userAgent.isNotBlank()) { "User agent cannot be blank" }
    }

    /**
     * Create a copy with updated API key
     */
    fun withApiKey(apiKey: String): CarespaceConfiguration {
        return copy(apiKey = apiKey)
    }

    /**
     * Create a copy with additional headers
     */
    fun withHeaders(headers: Map<String, String>): CarespaceConfiguration {
        return copy(defaultHeaders = defaultHeaders + headers)
    }
}